include/deal.II/multigrid/mg_coarse.h

```diff
@@ -329,6 +339,22 @@ MGCoarseGridApplySmoother<VectorType>::initialize (const MGSmootherBase<VectorTy
     (&coarse_smooth_,typeid(*this).name());
 }
 
+
+template<class VectorType>
+void
+MGCoarseGridApplySmoother<VectorType>::clear()
+{
+  coarse_smooth = nullptr;
+}
+
+
+template<class VectorType>
+MGCoarseGridApplySmoother<VectorType>::~MGCoarseGridApplySmoother()
+{
+  clear();
```

Comment:
Why do you need this destructor? Isn't the destructor of `SmartPointer` already doing everything? (Or rather, nothing, since it does not handle the memory?)